openmsupply/application-manager-web-app
# #708 consolidator changes request and updates by reviewer in review page

Fixes #708 

### Based on PR
[#646-#721-and-other-style-changes](#747 
With changes merged from PR [#741](#746) 

### Back-end branch
Using linked back-end of [#741F](#741F create locked review when assignment is locked)

### Changes
- Please ignore files `src/containers/Review/ReviewSubmit.tsx` and `src/utils/messages.ts` which are changes from #741 
- Added  element of **consolidation response** of change requested to lower level on `ReviewApplicantResponse`
- After the change is done, so now `isNewReviewResponse` is defined use the `shouldDim` to grey out the Consolidator changes request to previous Reviewer review response.

### Screenshots
- Also display on the Review Home the number of changes to be updated by Reviewer:
<img width="765" alt="Screen Shot 2021-05-31 at 1 00 47 PM" src="https://user-images.githubusercontent.com/16461988/120128556-65ec8300-c216-11eb-83c8-ad6822903831.png">
- Although on the list is still showing only **Update** - let me know what do you think:

![Screen Shot 2021-05-31 at 1 00 39 PM](https://user-images.githubusercontent.com/16461988/120128744-de534400-c216-11eb-8837-086bd110d03f.png)

- On the Review Page show the label **Update** on each Review with changes requested not updated:
<img width="666" alt="Screen Shot 2021-05-31 at 1 01 04 PM" src="https://user-images.githubusercontent.com/16461988/120128493-3a699880-c216-11eb-9234-48b74218ef71.png">

- After is updated will show the Consolidation greyed out and the Edit icon (as others Review responses with decision):
<img width="630" alt="Screen Shot 2021-05-31 at 1 40 07 PM" src="https://user-images.githubusercontent.com/16461988/120128498-3dfd1f80-c216-11eb-8cc0-54fca0033b9f.png">

- Something that I don't like is that after submitting the updated review, the Consolidator is showing (related to previous reveiwResponse) and that's misleading:
<img width="660" alt="Screen Shot 2021-05-31 at 1 40 20 PM" src="https://user-images.githubusercontent.com/16461988/120128499-3e95b600-c216-11eb-96ed-33ef2eeee515.png">
Some solutions to that:
- Show the **Updated** icon on the reviews with new ReviewResponse
- Don't show the Consolidator below the ReviewResponse after it's updated (just the Review history icon)
- Show the previous ReviewResponse on the Consolidation response box 

### Tests 
- Use Application serial: 45678 and logged in as **Consolidator2**
- Change some of the decision to "Disagree" and submit a Changes Request to **Reviewer1**
- Login as Reviewer1 and see what is to be updated, do the update and resubmit.
- You can also see the **Consolidator2** received updated changes on their Review page - no other indication other that a button to **Review** showing on the interface though.
